crates/burn-tensor/src/tensor/activation/base.rs

```diff
@@ -13,6 +13,32 @@ pub fn gelu<const D: usize, B: Backend>(tensor: Tensor<B, D>) -> Tensor<B, D> {
     Tensor::from_primitive(B::gelu(tensor.primitive))
 }
 
+/// Applies Parametric ReLu activation
+/// ` PReLu(x) = max(0,x) + \alpha * min(0,x)`
+pub fn prelu<const D: usize, B: Backend>(
+    tensor: Tensor<B, D>,
+    alpha: Tensor<B, 1>,
+) -> Tensor<B, D> {
+    let weight = if alpha.dims()[0] == 1 {
+        alpha.reshape([1; D])
+    } else if D >= 2 {
+        let channels = tensor.dims()[1];
+        let num_weights = alpha.dims()[0];
+        assert_eq!(
+            num_weights, channels,
+            "Number of channels and weights must be equal. Got no. of channels: {}, no. of weights: {}",
+            channels, num_weights
+        );
```

Comment:
I would use the TensorCheck API to do that, so that we can have a constant formatting for tensor errors. 
You can look at how it is done here: https://github.com/tracel-ai/burn/blob/4427768570503fcb3c437a2be32af640bb34a6fc/crates/burn-tensor/src/tensor/api/numeric.rs#L19 and here: https://github.com/tracel-ai/burn/blob/4427768570503fcb3c437a2be32af640bb34a6fc/crates/burn-tensor/src/tensor/api/check.rs#L52

But from what I understand, you need the tensor to have a dimension with the same number of items as the alpha tensor, so:

```
tensor: [batch_size, channel, dim]
alpha: [channel],
```

However, in your code you assume that the channel is always the second dimension, so:

```
tensor = [batch_size, channel, ..., dim];
```

I think I would add that documentation on the PRelU module, just specifying the shape expectations.

Also, just adding a comment that you are reshaping the alpha tensor to `[1, channel, ...]` for broadcasting can help.

Reply:
I don't see which api function can be used because, there is no function which has a check like the one I'm using. I am checking the number of channels with the shape of the weight to be learnt `alpha`